src/backend/distributed/connection/shared_connection_stats.c

```diff
@@ -587,6 +589,86 @@ SharedConnectionStatsShmemInit(void)
 }
 
 
+/*
+ * AdaptiveConnectionManagementFlag returns the appropriate connection flag
+ * for any given activeConnectionCount to remote nodes.
+ *
+ * This function should only be called if the code-path is capable of handling
+ * optional connections.
+ */
+int
+AdaptiveConnectionManagementFlag(int activeConnectionCount)
+{
+	if (UseConnectionPerPlacement())
+	{
+		/*
+		 * User wants one connection per placement, so no throttling is desired
+		 * and we do not set any flags.
+		 *
+		 * The primary reason for this is that allowing multiple backends to use
+		 * connection per placement could lead to unresolved self deadlocks. In other
+		 * words, each backend may stuck waiting for other backends to get a slot
+		 * in the shared connection counters.
+		 */
+		return 0;
```

Comment:
What does 0 mean? Using a macro, or something to remove the mapping could be useful